### ISSUE TITLE
[RFC] What version should the Docker latest tag point to?

### ISSUE THREAD

**Issue by hayley-jean** [EventStore/EventStore#2940]
**Context**

The `:latest` tag in [DockerHub](https://hub.docker.com/r/eventstore/eventstore) currently points to 20.10.2, which is the latest LTS release. However, the latest stable release is 21.2.0.

While using `:latest` as your image when deploying to production is against best docker practices, we should still decide on what version `:latest` will point to.

**Proposed Decision**

We can either do something similar to [dotnet](https://hub.docker.com/_/microsoft-dotnet-aspnet/):
- `:latest` points to the latest stable release (21.2.0 in our case)
- `:lts` points to the latest LTS release (20.10.2)

Or we can do something similar to [Ubuntu](https://hub.docker.com/_/ubuntu):
- `:latest` points to the latest LTS (20.10.2) because this is the version we recommend using in production
- `:rolling` points to the latest release regardless of LTS status (21.2.0).

**Additional Context**

If anyone feels strongly about this, or has other suggestions, please leave a comment on this issue

**Comment by oskardudycz** [comment]
Initially, I thought that I prefer the first option, as it's more straightforward and clear what `:latest` mean. However, I see that our scenario is closer to Ubuntu, as:
- New features are not back-ported to LTS.
- Bugs are generally not forward-ported to the newer version.
- We recommend using LTS and switching to the new versions when they need a specific new feature.
- 
Having that, plus knowing that people might not follow the best practices for not using the specific tag on production, I'm voting for the second option.

**Comment by Gutek** [comment]
my two cents. as a product for me it would be nice to not to accidentally slip into not supported version. so I would root for `:latest` pointing to LTS.

as a user however of the docker I usually use `:latest` when I really want a latest release (not alpha, beta but latest version). In other cases I'm searching for a different tag. This is helpful with development and trying new things. I would have personally problem for going to docker hub trying to figure out what version should I use to have latest stable version. 

Personally I also discourage usage of latest tag in teams - had few issues on production because of that. I would rather have specific version of ES docker image pushed to my team private repo. And only that image would be allowed to be used in production/staging.

so to summarize: I will understand if you will go with option `:latest` pointing to LTS, however I'm rooting for `:latest` pointing to latest.

**Comment by asbjornu** [comment]
Why not introduce a `:stable` tag pointing at the LTS and then have `:latest` pointing to the latest release? That would be the solution of least surprise to me, at least.

**Comment by nordfjord** [comment]
There's a third option that could probably be discussed: **Don't have :latest at all**

`:20` would point to the latest `20.x` release
`:20.10` would point to the latest `20.10.x` release
... etc

That would remove the footgun entirely

**Comment by alexeyzimarev** [comment]
I think it's good practice to have the `latest` tag.

**Comment by ramonsmits** [comment]
Tag `latest` IMHO does not have any guarantees and should not be used in production. We (Particular) push  public builds to a specific version and update `latest` "greatest" major.minor.patch but as we apply SemVer we recommend customers to not use `latest` but pin to a specific major. This ensures that customers can always safely upgrade to the latest image might a major or major.minor get updated.

**Comment by ahjohannessen** [comment]
HAProxy uses something similar to that @nordfjord suggested.

**Comment by ahjohannessen** [comment]
The following is from [HAProxy](https://hub.docker.com/_/haproxy) and seems reasonable:

- 2.4-dev19, 2.4-dev
- 2.4-dev19-alpine, 2.4-dev-alpine
- 2.3.10, 2.3, latest
- 2.3.10-alpine, 2.3-alpine, alpine
- 2.2.14, 2.2, lts

If I want 2.3.x I use `2.3` or if I want 2.2.x I use `2.2` - [docker-library-bot](https://github.com/docker-library-bot) updates tags as done [here]( https://github.com/docker-library/haproxy/commit/db9335048c43b78fb4daec1ac9d7d171fc209d78).

**Comment by hayley-jean** [comment]
Thanks for the suggestions everyone! We have decided to go for a similar approach to HAProxy:

1. `latest` points to the latest stable version of the most recent major release. This could be either an interim version or the LTS.
2. `lts` points to the latest version of the most recent LTS release.
3. The major version tag will point to the most recent release for that version. Currently, these will be `20.10` and `21.6`.
4. The version-specific tags such as `21.6.0-buster-slim` will still be published with each release.
5. Nightly tags will still be published to the container registry in the EventStore repo.
